src/components/views/context_menus/MessageContextMenu.tsx

```diff
@@ -281,6 +284,30 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
         this.closeMenu();
     };
 
+    private onReplyInThreadClick = (): void => {
+        const { mxEvent } = this.props;
+
+        if (!localStorage.getItem("mx_seen_feature_thread")) {
+            localStorage.setItem("mx_seen_feature_thread", "true");
+        }
+
+        if (mxEvent.getThread() && !mxEvent.isThreadRoot) {
+            showThread({
+                rootEvent: mxEvent.getThread().rootEvent,
+                initialEvent: mxEvent,
+                scroll_into_view: true,
+                highlighted: true,
+                push: RightPanelStore.instance.isOpen,
```

Comment:
Sidestepping the circular dep issue, but I noticed that the other implementation of open thread in MessageActionBar uses CardContext instead of RightPanelStore to set the push property. Could you extract the click handler from ReplyInThreadButton in MessageActionBar and share the logic? It makes sense that the two versions of the button behave exactly the same, and as a bonus you get around the circular dependency

Reply:
I am not sure I follow - how would I use the `CardContext` in the click handler? What am I missing? :sweat_smile:

Reply:
I'm assuming the button you're adding should behave the same as the one in the `MessageActionBar`? I'm suggesting you use exactly the same logic as here: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/views/messages/MessageActionBar.tsx#L182-L206
Either by copying it, or better by abstracting it out of `MessageActionBar` into a util and reusing it in both places. 

(Disclaimer: I'm not that familiar with how the panels work, so I'm just assuming if the `CardContext.isCard` logic works in `MessageActionBar` it should work in the right click menu)